zkvm/src/vm.rs

```diff
@@ -321,12 +322,12 @@ where
         Ok(())
     }
 
-    fn range(&mut self, i: u8) -> Result<(), VMError> {
-        if i < 1 || i > 64 {
+    fn range(&mut self, i: BitRange) -> Result<(), VMError> {
+        if !i.is_valid() {
```

Comment:
No need for this check - it's already encoded in the BitRange type. See the comment above.